### PR TITLE
Add hash to benchmarks

### PR DESCRIPTION
**Pull request by boris-kap**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: core/src/main/java/com/yahoo/oak/HashChunk.java [hunk 1]
```diff
@@ -332,6 +332,11 @@ HashChunkIter chunkIter(ThreadContext ctx) {
         }
 
 
+        /**
+         * find the index of the next occupied entry in the entryies array
```
Comment (sanastas) [inline core/src/main/java/com/yahoo/oak/HashChunk.java:336]: The explanation should say the "first" index, this is what name of the function suggests...

File: core/src/main/java/com/yahoo/oak/EntryHashSet.java [hunk 2]
```diff
@@ -665,7 +675,14 @@ boolean isEntryDeleted(ValueBuffer tempValue, int ei) {
         return tempValue.getSlice().isDeleted() != ValueUtils.ValueResult.FALSE;
     }
 
-
+    @Override
+    int getNextNonZeroIndex(int currentIndex) {
+        int nxtIdx = mapOfCleanEntries.nextSetBit(currentIndex + 1);
```
Comment (sanastas) [inline core/src/main/java/com/yahoo/oak/EntryHashSet.java:685]: Why do you think it is OK not to use synchronization while reading?

File: core/src/main/java/com/yahoo/oak/EntryArray.java [hunk 3]
```diff
@@ -56,6 +56,7 @@
      */
     protected static final int KEY_REF_OFFSET = 0;
     protected static final int VALUE_REF_OFFSET = 1;
+    protected static final int OPT_FIELD_OFFSET = 2;
```
[the remark this answers is not in the document]
Comment (boris-kap) [reply core/src/main/java/com/yahoo/oak/EntryArray.java:59]: this is the field after the KEY and VALUE fields, so it is always 2, regardless of the additionalFields Count, isn't it?

File: core/src/main/java/com/yahoo/oak/EntryHashSet.java (hunk 2 above)
Comment (boris-kap) [reply core/src/main/java/com/yahoo/oak/EntryHashSet.java:685]: if several processes read read the value at the same time, could it be a problem?

File: core/src/main/java/com/yahoo/oak/EntryArray.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (sanastas) [reply core/src/main/java/com/yahoo/oak/EntryArray.java:59]: If additionalFieldsCount is zero then there is no meaning in OPT_FIELD_OFFSET. Maybe, then OPT_FIELD_OFFSET can be 0 or somehow eliminated...

File: core/src/main/java/com/yahoo/oak/EntryHashSet.java (hunk 2 above)
Comment (sanastas) [reply core/src/main/java/com/yahoo/oak/EntryHashSet.java:685]: Generally, shouldn't be, but it depends how `BitSet` is implemented. If the update is more than a machine word, such an unprotected read may read partially updated structure. Do we know at least approximately whether BitSet is array with relevant bits set or may be a linked list with nodes, for example?

File: core/src/main/java/com/yahoo/oak/EntryArray.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (boris-kap) [reply core/src/main/java/com/yahoo/oak/EntryArray.java:59]: OPT_FIELD_OFFSET is the offset of the optional fields, which is always 2, in our case. Never 0.

File: core/src/main/java/com/yahoo/oak/EntryHashSet.java (hunk 2 above)
Comment (boris-kap) [reply core/src/main/java/com/yahoo/oak/EntryHashSet.java:685]: No, we don't know how it is implemented, we don't want to guess either. So, the synchronisation on function level is not enough, we need to add a lock and acquire it before every write/read on the bitset

**Pull request merged**